### ISSUE TITLE
TSQL:  Table Options Not Parsing Correctly

### ISSUE THREAD

**Issue by nathan-boyd** [antlr/grammars-v4#611]
The following SQL query is not parsing in the TSQL antlr4 grammar, but I verified that its correct by running it in SSMS.

```sql
CREATE TABLE dbo.TestTable (
    TableID uniqueidentifier NOT NULL,
    Value nvarchar(64) NOT NULL
    CONSTRAINT PK_TestTable_ID PRIMARY KEY (TableID)) WITH (DATA_COMPRESSION = PAGE)
```
running the test rig yields the following errors:
line 4:63 no viable alternative at input 'WITH('
line 4:64 no viable alternative at input 'DATA_COMPRESSION'

![antlr4_parse_tree](https://cloud.githubusercontent.com/assets/1316405/25238540/b9dee814-25bb-11e7-9c9f-c5b7c9c51221.png)
I'll work on the grammar and see if I can get it working, I think the problem is somewhere in the index_options

https://github.com/antlr/grammars-v4/blob/master/tsql/tsql.g4#L688-L696

```
// https://msdn.microsoft.com/en-us/library/ms188066.aspx
table_constraint
    : (CONSTRAINT id)?
       ((PRIMARY KEY | UNIQUE) clustered? '(' column_name_list (ASC | DESC)? ')' index_options? (ON id)?
       | CHECK (NOT FOR REPLICATION)? '(' search_condition ')')
    ;

index_options
    : WITH '(' index_option (',' index_option)* ')'
    ;
```

**Comment by zhongxiao37** [comment]
First, you missed one comma in the 3rd line. 
For the issue you described, the `create_table` rule missed `WITH` section. Need a fix in grammar file.

**Comment by nathan-boyd** [comment]
@zhongxiao37 the create table snippet in the description is syntactically correct, it parses in SSMS fine. Is there a tool you are using where this is not the case?

I submitted a fix in PR #615
